File: backend/shared/uidl/src/main/java/io/mateu/uidl/data/HorizontalLayout.java
package io.mateu.uidl.data;

import io.mateu.uidl.fluent.Component;
import java.util.List;
import lombok.Builder;

@Builder
public record HorizontalLayout(
    String id,
    List<Component> content,
    boolean spacing,
    boolean padding,
    boolean margin,
    SpacingVariant spacingVariant,
    HorizontalLayoutVerticalAlignment verticalAlignment,
    HorizontalLayoutJustification justification,
    boolean wrap,
    List<Integer> flexGrows,
    boolean fullWidth,
    String style,
    String cssClasses)
    implements Component {

  public HorizontalLayout(List<Component> content) {
    this(null, content, false, false, false, null, null, null, false, List.of(), false, null, null);
  }
}

File: frontend/web/vaadin/src/mateu/shared/apiClients/dtos/componentmetadata/HorizontalLayout.ts
import ComponentMetadata from "@mateu/shared/apiClients/dtos/ComponentMetadata";

export enum SpacingVariant {
    xs = "xs",
    s = "s",
    m = "m",
    l = "l",
    xl = "xl",
}

export enum HorizontalLayoutJustification {
    START = "START",
    CENTER = "CENTER",
    END = "END",
    BETWEEN = "BETWEEN",
    AROUND = "AROUND",
    EVENLY = "EVENLY",
}

export enum VerticalAlignment {
    STRETCH = "STRETCH",
    START = "START",
    CENTER = "CENTER",
    END = "END",
    BASELINE = "BASELINE",
}

export default interface HorizontalLayout extends ComponentMetadata {

    padding: boolean
    spacing: boolean
    fullWidth: boolean
    flexGrows: number[] | undefined
    wrap: boolean
    spacingVariant: SpacingVariant | undefined
    justification: HorizontalLayoutJustification | undefined
    verticalAlignment: VerticalAlignment | undefined

}
